### ISSUE TITLE
OB BROKE MASTER

### ISSUE THREAD

**Issue by procrasti** [orionblastar/K666#20]
Sorry everyone, OB accidentally checked in some changes to master, which have temporarily broken the build, we are working on this issue, will do our best not to break the build in the future, and will restore normal operation shortly.
- [x] merge in the changes from messages-fixed into messages
- [x] check that it runs according to the README.me, from git clone to localhost:8000 (hint, blow away dist, env, build and k666.egg-info and rerun k666-env.)
- [x] and all the changes are in origin/messages
- [x] merge messages into master
- [ ] double check master runs from clone to localhost according to the README.

**Comment by procrasti** [comment]
* [ ] merge in the changes from messages-fixed into messages
* [ ] check that it runs according to the README.me, from git clone to localhost:8000 (hint, blow away dist, env, build and k666.egg-info and rerun k666-env.)
* [ ] and all the changes are in messages
* [ ] merge messages into master
* [ ] double check master runs from clone to localhost according to the README.

**Comment by procrasti** [comment]
Hey don't worry too much though... I did a push for messages-fixes, and it pushed the local commits I'd made to messages too, even though I didn't mean to.

Difference between

`$ git push`

and

`$ git push -u origin <branchname>`

**Comment by orionblastar** [comment]
I removed the django messages references in master and tested that it ran. So master should be back to before I made the changes.

I merged messages-fixed into messages. I ran k666-env and it sends a message to myself.

I'm going to look into seeing if messages is fixed and meets all requirements.

**Comment by orionblastar** [comment]
Ok I deleted dist, env, build, and k666.egg-info and reran it. Got some warnings about zip and etc being depriciated, some directories missing. 

DEPRECATION: 'pip zip' and 'pip unzip` are deprecated, and will be removed in a future release.

Django-1.8.13/egg-dist-tmp-YMuox8
warning: no previously-included files matching '**pycache**' found under directory '_'
warning: no previously-included files matching '_.py[co]' found under directory '*'

It runs but has those warning/error messages.

Could we be missing the cache option and that is why it runs slower in dynamic than static?

Changes checked into messages from messages-fixed.

**Comment by procrasti** [comment]
Yep, warnings I can live with...

I'm not 100% sure of any of it.

I had to use pip unzip to get django-messages to play nice... without it you'll see an error it can't find django-messages/migrations or something like that... It's not zip safe, and they haven't set it in their setup.py... So, django-messages is basically broken... BUT I've raised an issue to deal with this problem separately... I think we'll deal with this in https://github.com/orionblastar/K666/issues/22.

I have no idea about those cache files and the compiled python files (.pyc and .pyo)... I assume they all get built at runtime, and should be ready from then on... but I guess I have to check what happens in production more clearly... This is not the caching I use on the main site, but caching compile python (.pyc) and python objects (I think a later build stage, .pyo files)...

Again... maybe raise an issue for it for investigation, I suspect they have performance implications, hopefully only once per deploy or something... but for now, they don't bother me.

BTW: Tick off the done items in the checklist that you believe you have done for full marks!

**Comment by procrasti** [comment]
You're handling a very steep learning curve btw, every issue you close basically teaches you a lot, and means a lot of the stuff you'll just do automatically as we continue on... but new concepts are going to be thrown at you more or less forever... it never gets easy.

**Comment by procrasti** [comment]
Ummm... did you do a legit merge with 'git merge', or did you do something else, like copy the files over?

I don't mind either way, just the commit log in messages doesn't contain the commit from messages-fixed.

No real problem I suppose, just curious.

Then again, typing it all in yourself gives you a much better idea than blindly merging stuff.

Pretty much I want you to study every commit made and why it is like that...

It's your code after all!

**Comment by orionblastar** [comment]
I just copied the files that got modified over. I was not aware of the git merge option or maybe I forgot it. I'm absentminded at times and forget stuff. It is due to my mental illness or the medicine I am on. The more I do something, I think the more likely I'll remember it.

**Comment by procrasti** [comment]
Yeah, no biggie... in general use git merge, not file copy or paste or whatever... acceptable if you type it in yourself maybe in order to learn...

git merge will keep track of the history properly... but no big problem, you're still learning.

BTW: Tick off the checkboxes at the head of this issue as you complete them please.

**Comment by orionblastar** [comment]
OK I had a conflict in index.html in static and setup.py in the root directory. I resolved them and committed it to Master and made a push. Merge for some reason didn't want to work. Maybe because I copied the files instead of merging them?

Here is the errors I get following the readme in Master. I think checking the build for the Readme should be on the list before merging into master. 

(env) norman@Lazarus0 ~/k666test $ . ./k666-env
/usr/local/bin/virtualenv
/home/norman/k6messages/env/bin/python
Running virtualenv with interpreter /home/norman/k6messages/env/bin/python
Using real prefix '/usr'
New python executable in /home/norman/k666test/env/bin/python
Installing setuptools, pip, wheel...done.
running install
running bdist_egg
running egg_info
creating k666.egg-info
writing requirements to k666.egg-info/requires.txt
writing k666.egg-info/PKG-INFO
writing top-level names to k666.egg-info/top_level.txt
writing dependency_links to k666.egg-info/dependency_links.txt
writing manifest file 'k666.egg-info/SOURCES.txt'
reading manifest file 'k666.egg-info/SOURCES.txt'
writing manifest file 'k666.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
warning: install_lib: 'build/lib.linux-x86_64-2.7' does not exist -- no Python modules to install

creating build
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying k666.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying k666.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying k666.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying k666.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying k666.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating dist
creating 'dist/k666-0.0.0-py2.7.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing k666-0.0.0-py2.7.egg
Copying k666-0.0.0-py2.7.egg to /home/norman/k666test/env/lib/python2.7/site-packages
Adding k666 0.0.0 to easy-install.pth file

Installed /home/norman/k666test/env/lib/python2.7/site-packages/k666-0.0.0-py2.7.egg
Processing dependencies for k666==0.0.0
Searching for Django<1.9rc1
Reading https://pypi.python.org/simple/Django/
Best match: Django 1.8.13
Downloading https://pypi.python.org/packages/85/0e/0200c2c3792a4fb8bda6a8597d267abaeb00d862e3a110f99cb101398d4d/Django-1.8.13.tar.gz#md5=a77e1ba9991f762de20bf03de57e39eb
Processing Django-1.8.13.tar.gz
Writing /tmp/easy_install-m3Ubki/Django-1.8.13/setup.cfg
Running Django-1.8.13/setup.py -q bdist_egg --dist-dir /tmp/easy_install-m3Ubki/Django-1.8.13/egg-dist-tmp-Ne5MPn
warning: no previously-included files matching '**pycache**' found under directory '_'
warning: no previously-included files matching '_.py[co]' found under directory '*'
creating /home/norman/k666test/env/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg
Extracting Django-1.8.13-py2.7.egg to /home/norman/k666test/env/lib/python2.7/site-packages
Adding Django 1.8.13 to easy-install.pth file
Installing django-admin.py script to /home/norman/k666test/env/bin
Installing django-admin script to /home/norman/k666test/env/bin

Installed /home/norman/k666test/env/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg
Searching for django-messages
Reading https://pypi.python.org/simple/django-messages/
Best match: django-messages 0.5.3
Downloading https://pypi.python.org/packages/cc/06/610bea47334e935de38356d3086caba677a43e942f89cc88d5d6d7577b95/django-messages-0.5.3.tar.gz#md5=50f905010465f9765263b4a70d566d19
Processing django-messages-0.5.3.tar.gz
Writing /tmp/easy_install-2eVqC3/django-messages-0.5.3/setup.cfg
Running django-messages-0.5.3/setup.py -q bdist_egg --dist-dir /tmp/easy_install-2eVqC3/django-messages-0.5.3/egg-dist-tmp-tO6XHi
zip_safe flag not set; analyzing archive contents...
Moving django_messages-0.5.3-py2.7.egg to /home/norman/k666test/env/lib/python2.7/site-packages
Adding django-messages 0.5.3 to easy-install.pth file

Installed /home/norman/k666test/env/lib/python2.7/site-packages/django_messages-0.5.3-py2.7.egg
Searching for django-allauth
Reading https://pypi.python.org/simple/django-allauth/
Best match: django-allauth 0.25.2
Downloading https://pypi.python.org/packages/94/81/e6314914b87409a224bcdfef55a8978b2bc19417db7e783499873a6e3634/django-allauth-0.25.2.tar.gz#md5=ffc7520bc897ed6c1880a12512d85baa
Processing django-allauth-0.25.2.tar.gz
Writing /tmp/easy_install-ivwOKv/django-allauth-0.25.2/setup.cfg
Running django-allauth-0.25.2/setup.py -q bdist_egg --dist-dir /tmp/easy_install-ivwOKv/django-allauth-0.25.2/egg-dist-tmp-MyHCSL
zip_safe flag not set; analyzing archive contents...
allauth.socialaccount.providers.facebook.locale: module references **file**
creating /home/norman/k666test/env/lib/python2.7/site-packages/django_allauth-0.25.2-py2.7.egg
Extracting django_allauth-0.25.2-py2.7.egg to /home/norman/k666test/env/lib/python2.7/site-packages
Adding django-allauth 0.25.2 to easy-install.pth file

Installed /home/norman/k666test/env/lib/python2.7/site-packages/django_allauth-0.25.2-py2.7.egg
Searching for requests>=1.0.3
Reading https://pypi.python.org/simple/requests/
Best match: requests 2.10.0
Downloading https://pypi.python.org/packages/49/6f/183063f01aae1e025cf0130772b55848750a2f3a89bfa11b385b35d7329d/requests-2.10.0.tar.gz#md5=a36f7a64600f1bfec4d55ae021d232ae
Processing requests-2.10.0.tar.gz
Writing /tmp/easy_install-ppOej4/requests-2.10.0/setup.cfg
Running requests-2.10.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-ppOej4/requests-2.10.0/egg-dist-tmp-Ofk2lv
warning: no files found matching 'test_requests.py'
creating /home/norman/k666test/env/lib/python2.7/site-packages/requests-2.10.0-py2.7.egg
Extracting requests-2.10.0-py2.7.egg to /home/norman/k666test/env/lib/python2.7/site-packages
Adding requests 2.10.0 to easy-install.pth file

Installed /home/norman/k666test/env/lib/python2.7/site-packages/requests-2.10.0-py2.7.egg
Searching for requests-oauthlib>=0.3.0
Reading https://pypi.python.org/simple/requests-oauthlib/
Best match: requests-oauthlib 0.6.1
Downloading https://pypi.python.org/packages/f9/98/a1aaae4bbcde0e98d6d853c4f08bd52f20b0005cefb881679bcdf7ea7a00/requests-oauthlib-0.6.1.tar.gz#md5=f159bc7675ebe6a2d76798f4c00c5bf8
Processing requests-oauthlib-0.6.1.tar.gz
Writing /tmp/easy_install-t_bG5X/requests-oauthlib-0.6.1/setup.cfg
Running requests-oauthlib-0.6.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-t_bG5X/requests-oauthlib-0.6.1/egg-dist-tmp-xcbAMP
no previously-included directories found matching '*.pyc'
creating /home/norman/k666test/env/lib/python2.7/site-packages/requests_oauthlib-0.6.1-py2.7.egg
Extracting requests_oauthlib-0.6.1-py2.7.egg to /home/norman/k666test/env/lib/python2.7/site-packages
Adding requests-oauthlib 0.6.1 to easy-install.pth file

Installed /home/norman/k666test/env/lib/python2.7/site-packages/requests_oauthlib-0.6.1-py2.7.egg
Searching for python-openid>=2.2.5
Reading https://pypi.python.org/simple/python-openid/
Best match: python-openid 2.2.5
Downloading https://pypi.python.org/packages/aa/85/0bac9096d208f04c9de9d49f3a4e27fec8bfd56ee841c185f5cf420a24d2/python-openid-2.2.5.zip#md5=f89d9d4f4dccfd33b5ce34eb4725f751
Processing python-openid-2.2.5.zip
Writing /tmp/easy_install-xHnj3r/python-openid-2.2.5/setup.cfg
Running python-openid-2.2.5/setup.py -q bdist_egg --dist-dir /tmp/easy_install-xHnj3r/python-openid-2.2.5/egg-dist-tmp-Hm7uTT
warning: no files found matching 'CHANGELOG'
warning: no files found matching '_.css' under directory 'doc'
warning: no files found matching '_.html' under directory 'doc'
zip_safe flag not set; analyzing archive contents...
Moving python_openid-2.2.5-py2.7.egg to /home/norman/k666test/env/lib/python2.7/site-packages
Adding python-openid 2.2.5 to easy-install.pth file

Installed /home/norman/k666test/env/lib/python2.7/site-packages/python_openid-2.2.5-py2.7.egg
Searching for oauthlib>=0.6.2
Reading https://pypi.python.org/simple/oauthlib/
Best match: oauthlib 1.1.2
Downloading https://pypi.python.org/packages/a8/11/fb71ced7057b2a8929f51959f4e97bcee9f687aaf896c521984e67118b90/oauthlib-1.1.2.tar.gz#md5=93db15cb6519a8707b75a141e865d86e
Processing oauthlib-1.1.2.tar.gz
Writing /tmp/easy_install-X1CdnI/oauthlib-1.1.2/setup.cfg
Running oauthlib-1.1.2/setup.py -q bdist_egg --dist-dir /tmp/easy_install-X1CdnI/oauthlib-1.1.2/egg-dist-tmp-zntYpk
zip_safe flag not set; analyzing archive contents...
Moving oauthlib-1.1.2-py2.7.egg to /home/norman/k666test/env/lib/python2.7/site-packages
Adding oauthlib 1.1.2 to easy-install.pth file

Installed /home/norman/k666test/env/lib/python2.7/site-packages/oauthlib-1.1.2-py2.7.egg
Finished processing dependencies for k666==0.0.0
ERROR: unknown command "unzip"
Skipping creation of NoticeTypes as notification app not found
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/**init**.py", line 354, in execute_from_command_line
    utility.execute()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/**init**.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/base.py", line 394, in run_from_argv
    self.execute(_args, *_cmd_options)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/base.py", line 445, in execute
    output = self.handle(_args, *_options)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/commands/migrate.py", line 93, in handle
    executor = MigrationExecutor(connection, self.migration_progress_callback)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/migrations/executor.py", line 19, in **init**
    self.loader = MigrationLoader(self.connection)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/migrations/loader.py", line 47, in **init**
    self.build_graph()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/migrations/loader.py", line 185, in build_graph
    self.load_disk()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/migrations/loader.py", line 93, in load_disk
    for name in os.listdir(directory):
OSError: [Errno 20] Not a directory: '/home/norman/k666test/env/local/lib/python2.7/site-packages/django_messages-0.5.3-py2.7.egg/django_messages/migrations'
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/**init**.py", line 354, in execute_from_command_line
    utility.execute()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/**init**.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/base.py", line 394, in run_from_argv
    self.execute(_args, *_cmd_options)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/contrib/auth/management/commands/createsuperuser.py", line 50, in execute
    return super(Command, self).execute(_args, *_options)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/base.py", line 445, in execute
    output = self.handle(_args, *_options)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/contrib/auth/management/commands/createsuperuser.py", line 81, in handle
    default_username = get_default_username()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/contrib/auth/management/**init**.py", line 177, in get_default_username
    auth_app.User._default_manager.get(username=default_username)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/models/manager.py", line 127, in manager_method
    return getattr(self.get_queryset(), name)(_args, *_kwargs)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/models/query.py", line 328, in get
    num = len(clone)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/models/query.py", line 144, in __len__
    self._fetch_all()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/models/query.py", line 965, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/models/query.py", line 238, in iterator
    results = compiler.execute_sql()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/models/sql/compiler.py", line 840, in execute_sql
    cursor.execute(sql, params)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/utils.py", line 98, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/backends/sqlite3/base.py", line 318, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: no such table: auth_user
Performing system checks...

System check identified no issues (0 silenced).
Unhandled exception in thread started by <function wrapper at 0x7f0505c24de8>
Traceback (most recent call last):
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/utils/autoreload.py", line 229, in wrapper
    fn(_args, *_kwargs)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/commands/runserver.py", line 116, in inner_run
    self.check_migrations()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/core/management/commands/runserver.py", line 168, in check_migrations
    executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/migrations/executor.py", line 19, in **init**
    self.loader = MigrationLoader(self.connection)
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/migrations/loader.py", line 47, in **init**
    self.build_graph()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/migrations/loader.py", line 185, in build_graph
    self.load_disk()
  File "/home/norman/k666test/env/local/lib/python2.7/site-packages/Django-1.8.13-py2.7.egg/django/db/migrations/loader.py", line 93, in load_disk
    for name in os.listdir(directory):
OSError: [Errno 20] Not a directory: '/home/norman/k666test/env/local/lib/python2.7/site-packages/django_messages-0.5.3-py2.7.egg/django_messages/migrations'

**Comment by procrasti** [comment]
I think this is the problem: 

> ERROR: unknown command "unzip"

Probably you are using an old version of pip...

Might need to add:

```
pip install --upgrade pip
```

Into k666-env.

Let me try it.

**Comment by procrasti** [comment]
It's worse than that... unzip is already removed from pip... need pip version less than 7...

so put in

```
pip install --upgrade 'pip<7'
```

This is terrible, but seen you have forked django-messages, good!

**Comment by procrasti** [comment]
Okay... did that... but doesn't fix the issue!

You removed django-settings and stuff from master when you tried to fix up the broken master problem... which is why I told you not to bother fixing it! You've merged in the changes from messages into master, but you didn't undo the changes to master you did undoing the accidental check in to master... So, you have to reapply that..

I'll let you fix master...

You can do:

```
git diff origin/messages
```

To see what you might have to do to get it working.

**Comment by orionblastar** [comment]
I checked master back in with a push. Removed all references to messages as per the diff. Master works again but has no link to a profile page, I guess we didn't add it yet but it is in the index.html but static and dynamic.

I know I'm in over my head here with the different versions changing things. If unzip was removed from pip what other stuff does it support? I remember 7-zip could do zip files. Does the new pip using 7-zip or anything else? http://computernetworkingnotes.com/ubuntu-12-04-tips-and-tricks/how-to-install-7zip-in-ubuntu.html

**Comment by orionblastar** [comment]
Looks like they removed unzip as of version 1.5 so we should use 1.4 instead?
https://pip.pypa.io/en/stable/news/

**Comment by procrasti** [comment]
No, it was version 7 when they removed zip and unzip.

What I checked in is fine.

You now need to look at the differences between master and messages, and manually bring master inline with messages, so that it too supports messages.

What happened was, you accidently checked in some changes from messages into master, but then you manually undid those changes from master, and when you later merged in the changes from messages, it did not undo those manual changes you made to master... as it should, because merge only brings changes, and you may have made other changes to master that were irrelevant to messages. Git cannot tell the difference.

So, you must manually fix up master to bring it inline with messages... I want messages working in master now. The fact that master is broken is secondary to this goal.

You can use:
`$ git diff origin/messages`

To determine what changes have to made to make it work like messages.

**Comment by procrasti** [comment]
What you should actually do, is checkout messages, merge master into it, then fix messages so it works properly, then push messages to origin, and once that works, merge messages into master.

```
$ git clone ....
$ git checkout messages
$ git merge master

# Fix messages up, so it works like you want it to

$ git add <changedfiles>
$ git commit

$ git push -u origin messages

$ git checkout master
$ git merge messages

# Check master works like you want it to, with messages

$ git push
```

**Comment by procrasti** [comment]
I think the main lesson to learn here is that before you merge a branch into master, first merge master into the branch, so that the branch has all the latest changes to master in it, and make sure it works before merging the branch back into master.

**Comment by procrasti** [comment]
Things to do now:
- [ ] Merge master into messages
- [ ] Fix the messages branch
- [ ] Merge messages back into master.

**Comment by orionblastar** [comment]
I seem to have some sort of problem cloning with git:
norman@Lazarus0 ~ $ git clone https://github.org/orionblastar/K666
Cloning into 'K666'...
fatal: unable to access 'https://github.org/orionblastar/K666/': Failed to connect to github.org port 443: Connection timed out
norman@Lazarus0 ~ $ git clone https://github.org/orionblastar/K666
Cloning into 'K666'...
fatal: unable to access 'https://github.org/orionblastar/K666/': Failed to connect to github.org port 443: Connection timed out

**Comment by procrasti** [comment]
Ummm yeah, you got to clone from the .git:
```
$ git clone https://github.org/orionblastar/K666.git
```

interestingly, k666 (lowercase K) also works!

**Comment by procrasti** [comment]
github.COM!!!

Crazy, I know... did it myself... bitbucket (the commercial one) is .org, but github (the open source only one) is .com!

wtf?

**Comment by orionblastar** [comment]
Ok I could not push to messages after I made the changes and tested that it worked.

(env) norman@Lazarus0 ~/K666 $ git checkout -b messages
fatal: A branch named 'messages' already exists.
(env) norman@Lazarus0 ~/K666 $ git checkout messages
Switched to branch 'messages'
Your branch is up-to-date with 'origin/messages'.
(env) norman@Lazarus0 ~/K666 $ git merge messages
Already up-to-date.
(env) norman@Lazarus0 ~/K666 $ git push -u messages
fatal: 'messages' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
(env) norman@Lazarus0 ~/K666 $ git checkout messages
Already on 'messages'
Your branch is up-to-date with 'origin/messages'.
(env) norman@Lazarus0 ~/K666 $ git push -b messages
error: unknown switch `b'
usage: git push [<options>] [<repository> [<refspec>...]]

```
-v, --verbose         be more verbose
-q, --quiet           be more quiet
--repo <repository>   repository
--all                 push all refs
--mirror              mirror all refs
-d, --delete          delete refs
--tags                push tags (can't be used with --all or --mirror)
-n, --dry-run         dry run
--porcelain           machine-readable output
-f, --force           force updates
--force-with-lease[=<refname>:<expect>]
                      require old value of ref to be at this value
--recurse-submodules[=<check|on-demand|no>]
                      control recursive pushing of submodules
--thin                use thin pack
--receive-pack <receive-pack>
                      receive pack program
--exec <receive-pack>
                      receive pack program
-u, --set-upstream    set upstream for git pull/status
--progress            force progress reporting
--prune               prune locally removed refs
--no-verify           bypass pre-push hook
--follow-tags         push missing but relevant tags
--signed[=<yes|no|if-asked>]
                      GPG sign the push
--atomic              request atomic transaction on remote side
-4, --ipv4            use IPv4 addresses only
-6, --ipv6            use IPv6 addresses only
```

(env) norman@Lazarus0 ~/K666 $ git push origin messages
Username for 'https://github.com': OrionBlastar
Password for 'https://OrionBlastar@github.com': 
Everything up-to-date
(env) norman@Lazarus0 ~/K666 $ 

(env) norman@Lazarus0 ~/K666 $ git commit -m "Fixing messages"
[master 2f0f0c7] Fixing messages
 4 files changed, 6 insertions(+), 1 deletion(-)
(env) norman@Lazarus0 ~/K666 $ git push -u origin messages
Username for 'https://github.com': OrionBlastar
Password for 'https://OrionBlastar@github.com': 
Branch messages set up to track remote branch messages from origin.
Everything up-to-date

(env) norman@Lazarus0 ~/K666 $ git push origin messages
Username for 'https://github.com': OrionBlastar
Password for 'https://OrionBlastar@github.com': 
Everything up-to-date
(env) norman@Lazarus0 ~/K666 $ git push
Username for 'https://github.com': OrionBlastar
Password for 'https://OrionBlastar@github.com': 
Counting objects: 10, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (10/10), done.
Writing objects: 100% (10/10), 950 bytes | 0 bytes/s, done.
Total 10 (delta 7), reused 0 (delta 0)
To http://github.org/orionblastar/K666
   0af2a0b..2f0f0c7  master -> master

I could not push into messages but I could into master.

I copied the URL from the project, why does it have github.org in it, I don't know.
Master should have messages fixes in it. Sorry but I could not merge with messages or push into it or anything, it kept saying it was up to date, but clearly it is not up to date.

**Comment by orionblastar** [comment]
I tried again: 
git clone https://github.com/orionblastar/K666.git k666fix
Cloning into 'k666fix'...
remote: Counting objects: 339, done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 339 (delta 5), reused 0 (delta 0), pack-reused 320
Receiving objects: 100% (339/339), 61.53 KiB | 0 bytes/s, done.
Resolving deltas: 100% (174/174), done.
Checking connectivity... done.
norman@Lazarus0 ~ $ cd k666fix
norman@Lazarus0 ~/k666fix $ git checkout messages
Branch messages set up to track remote branch messages from origin.
Switched to a new branch 'messages'
norman@Lazarus0 ~/k666fix $ git merge messages
Already up-to-date.
norman@Lazarus0 ~/k666fix $ git status
On branch messages
Your branch is up-to-date with 'origin/messages'.
nothing to commit, working directory clean

**Comment by procrasti** [comment]
git merge master, not git merge messages...

Anyway... it looks like you've merged master into messages already, no?

So, you've lost all the changes in messages that made messages work... so umm, you have to do them all again!

Be careful with git... sometimes I guess you can push your changes to the remote repository too early... cause you could have gotten the files you needed with:

```
   $ git diff origin/messages # to see what has changed after merging in master.
   $ git checkout --theirs origin/messages k666/settings.py k666/urls.py and index.html, common_links, etc... # to get the files you want that make messages work...
```

**Comment by procrasti** [comment]
Pretty sure this is done, right?

Ironically, pretty sure messages doesn't work right now but master has messages in it, not sure exactly what's going on... just merge latest changes into messages... but it's working, so all good!

**Comment by orionblastar** [comment]
I tried to merge with messages but git keeps telling me that it is up to date and doesn't need changing. I don't know why it does that.

norman@Lazarus0 ~/K666 $ git merge origin messages
Already up-to-date.

But messages does not have the changes in it that master has.

I must be getting confused with the git commands or maybe there is a glitch or something in git or github.

**Comment by procrasti** [comment]
I've already done it... messages and master are up to date.

It's why I closed the issue.
